### PR TITLE
Remove binding of $this to group()

### LINKED ISSUE #2559: Remove binding of $this to group()
As we've supported `$app` as a parameter to `group()`'s callback, we should remove the binding of `$this` to the closure.

### PR DESCRIPTION
**Pull request by vaga**
It's my first contribution. :tada: 

Closes #2559

### CODE REVIEW

File: tests/AppTest.php [hunk 1]
```diff
@@ -368,6 +368,16 @@ public function testEmptyRoute()
     /********************************************************************************
      * Route Groups
      *******************************************************************************/
+    public function testGroupWithBindingOfThis()
```
Comment (akrabat) [inline tests/AppTest.php]: Please rename to `testGroupClosureIsBoundToThisClass`

**Review by akrabat — CHANGES_REQUESTED** [review]
Just a minor nitpick on a test name.

Also, please update the CHANGELOG

**Comment by mbohal** [comment]
@akrabat Should I update the https://github.com/slimphp/Slim-Website/tree/gh-pages-4.x to reflect this change? 
I'm not sure what is going to happen with the docs when Slim 4 gets released and I do not want to introduce any conflicts.

**Comment by akrabat** [comment]
@mbohal Raise a PR against that branch please.

**Comment by akrabat** [comment]
Thanks @vaga !

**Pull request merged**